sklearnex/neighbors/common.py

```diff
@@ -71,9 +66,12 @@ def _fit_validation(self, X, y=None):
                 self.effective_metric_ = "chebyshev"
 
         if not isinstance(X, (KDTree, BallTree, sklearn_NeighborsBase)):
-            self._fit_X = self._check_array(
-                X, dtype=[np.float64, np.float32], accept_sparse=True
-            )
+            if not hasattr(X, "__sycl_usm_array_interface__"):
+                self._fit_X = _check_array(
+                    X, dtype=[np.float64, np.float32], accept_sparse=True
+                )
+            else:
+                self._fit_X = X
```

Comment:
This is really looks like not a good workaround.
Try avoid checking this attribute for input data.
We have to use only numpy arrays internally now.